test/integration/smoke/test_outofbandmanagement.py

```diff
@@ -0,0 +1,561 @@
+
+        try:
+            self.issuePowerActionCmd('STATUS')
+            self.fail("Exception was expected, oobm is disabled at host level")
+        except Exception: pass
+
+        # Enable at host level
+        cmd = enableOutOfBandManagementForHost.enableOutOfBandManagementForHostCmd()
+        cmd.hostid = host.id
+        response = self.apiclient.enableOutOfBandManagementForHost(cmd)
+
+        response = self.issuePowerActionCmd('STATUS')
+        self.assertEqual(response.powerstate, 'Off')
+
+
+    @attr(tags=["smoke", "advanced"])
+    def test_oobm_issue_power_status(self):
+        """
+            Tests out-of-band management issue power action
+        """
+        self.configureAndEnableOobm()
+        self.startIpmiServer()
+        bmc = IpmiServerContext().bmc
+        bmc.powerstate = 'on'
+        response = self.issuePowerActionCmd('STATUS')
+        self.assertEqual(response.powerstate, 'On')
+
+
+    @attr(tags=["smoke", "advanced"])
+    def test_oobm_issue_power_on(self):
+        """
+            Tests out-of-band management issue power on action
+        """
+        self.configureAndEnableOobm()
+        self.startIpmiServer()
+        self.issuePowerActionCmd('ON')
+        response = self.issuePowerActionCmd('STATUS')
+        self.assertEqual(response.powerstate, 'On')
+
+
+    @attr(tags=["smoke", "advanced"])
+    def test_oobm_issue_power_off(self):
+        """
+            Tests out-of-band management issue power off action
+        """
+        self.configureAndEnableOobm()
+        self.startIpmiServer()
+        self.issuePowerActionCmd('OFF')
+        response = self.issuePowerActionCmd('STATUS')
+        self.assertEqual(response.powerstate, 'Off')
+
+
+    @attr(tags=["smoke", "advanced"])
+    def test_oobm_issue_power_cycle(self):
+        """
+            Tests out-of-band management issue power cycle action
+        """
+        self.configureAndEnableOobm()
+        self.startIpmiServer()
+        self.issuePowerActionCmd('CYCLE')
+        response = self.issuePowerActionCmd('STATUS')
+        self.assertEqual(response.powerstate, 'On')
+
+
+    @attr(tags=["smoke", "advanced"])
+    def test_oobm_issue_power_reset(self):
+        """
+            Tests out-of-band management issue power reset action
+        """
+        self.configureAndEnableOobm()
+        self.startIpmiServer()
+        self.issuePowerActionCmd('RESET')
+        response = self.issuePowerActionCmd('STATUS')
+        self.assertEqual(response.powerstate, 'On')
+
+
+    @attr(tags=["smoke", "advanced"])
+    def test_oobm_issue_power_soft(self):
+        """
+            Tests out-of-band management issue power soft action
+        """
+        self.configureAndEnableOobm()
+        self.startIpmiServer()
+        self.issuePowerActionCmd('SOFT')
+        response = self.issuePowerActionCmd('STATUS')
+        self.assertEqual(response.powerstate, 'Off')
```

Comment:
The ``test_oobm_issue``* methods seem to be very similar.  Have consideration been given to extracting the commonalities to a template method?